src/PathProcessor/CivicrmPathProcessor.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Drupal\civicrm\PathProcessor;
+
+use Drupal\Core\PathProcessor\InboundPathProcessorInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+use Drupal\civicrm\Civicrm;
+
+class CivicrmPathProcessor implements InboundPathProcessorInterface {
+
+  public function processInbound($path, Request $request) {
+    // if the path is a civicrm path   
+    if (strpos($path, '/civicrm/') === 0) {
+      // initialize civicrm
+      $civicrm = new Civicrm();
+      // fetch civicrm menu items
+      $items = \CRM_Core_Menu::items();
+      foreach (array_keys($items) as $item) {
+        // if he current path is a civicrm path
+        if(('/' . $item  . '/') == $path) {
```

Comment:
Maybe I'm reading it wrong but substituting values for variables this statement looks like `if ('/civicrm/ajax/menujs/1/en_US/1/foo' == '/civicrm/ajax/menujs/')` which would always evaluate to `false`.